editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/FuseIntegrationProjectCreatorRunnable.java

```diff
@@ -111,6 +111,14 @@ public void run(IProgressMonitor monitor) throws InvocationTargetException, Inte
 			}
 		}
 
+		try {
+			Display.getDefault().syncExec( () -> postProjectCreationSetup(prj, subMonitor));
+		} finally {
+			subMonitor.done();
+		}
+	}
+
+	private void postProjectCreationSetup(IProject prj, SubMonitor subMonitor) {
```

Comment:
better to use IMonitor as parameter and then call SubMonitor.convert to create a new Submonitor otherwise it means that you would have to track the numbers of operations across function call